scripts/build/TestImpactAnalysis/Testing/test_tiaf_unit_tests.py

```diff
@@ -0,0 +1,233 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+
+from logging import getLogger
+import os
+import json
+from typing import Counter
+from tiaf import TestImpact
+from tiaf_driver import main
+import pytest
+
+logging = getLogger("tiaf")
+
+
+def assert_list_content_equal(list1, list2):
+    assert Counter(list1) == Counter(list2)
+
+
+class TestTIAFUnitTests():
```

Comment:
A small note/suggestion that you can divide unit tests further (i.e. on a function level). This allows shared setup and decorators, especially for mocking. For instance, there seemed to be a few functions that call S3 that is mocked. If that function has its own testing class, you can use one single mock decorator on the class and all test functions will inherit it.

Reply:
Thanks, will do so!

Reply:
After some exploration, implementing this introduced some odd issues with the existing fixtures I have written, so I'm going to stick to using pytest mock rather than mixing in unittest.mock with pytest-mock.